src/pepr/operator/controllers/network/generators/kubeAPI.ts

```diff
@@ -41,82 +56,107 @@ export function kubeAPI() {
     log.error({ err }, msg);
   }
 }
 
 /**
- * When the kubernetes Service is created or updated, update the API server CIDR
- * @param svc The Service for the API server
+ * When the Kubernetes Service is created or updated, update the API server CIDR.
+ *
+ * If a static CIDR is defined, it skips fetching the EndpointSlice and uses the static value.
+ *
+ * @param {kind.Service} svc - The Service object for the API server.
  */
 export async function updateAPIServerCIDRFromService(svc: kind.Service) {
   try {
-    log.debug(
-      "Processing watch for api service, getting endpoint slices for updating API server CIDR",
-    );
-    const slice = await K8s(kind.EndpointSlice).InNamespace("default").Get("kubernetes");
-    await updateAPIServerCIDR(slice, svc);
+    if (UDSConfig.kubeApiCidr) {
+      log.debug("Processing watch for api service, using configured API CIDR for endpoints");
+      await updateAPIServerCIDR(svc, UDSConfig.kubeApiCidr);
+    } else {
+      log.debug(
+        "Processing watch for api service, getting endpoint slices for updating API server CIDR",
+      );
+      const slice = await K8s(kind.EndpointSlice).InNamespace("default").Get("kubernetes");
+      await updateAPIServerCIDR(svc, slice);
+    }
   } catch (err) {
-    const msg = "Failed to update network policies from api service watch";
+    const msg = "Failed to update network policies from API service watch";
     log.error({ err }, msg);
   }
 }
 
 /**
- * Update the API server CIDR and update the NetworkPolicies
+ * Update the API server CIDR and apply it to the NetworkPolicies.
  *
- * @param slice The EndpointSlice for the API server
- * @param svc The Service for the API server
+ * @param {kind.Service} svc - The Service object representing the Kubernetes API server.
+ * @param {kind.EndpointSlice | string} slice - Either the EndpointSlice for dynamic CIDR generation or a static CIDR string.
  */
-export async function updateAPIServerCIDR(slice: kind.EndpointSlice, svc: kind.Service) {
-  const { endpoints } = slice;
+export async function updateAPIServerCIDR(svc: kind.Service, slice: kind.EndpointSlice | string) {
   const k8sApiIP = svc.spec?.clusterIP;
 
-  // Flatten the endpoints into a list of IPs
-  const peers = endpoints?.flatMap(e => e.addresses);
+  let peers: string[] = [];
 
+  // Handle static CIDR or dynamic EndpointSlice
+  if (typeof slice === "string") {
+    peers.push(slice);
+  } else {
+    const { endpoints } = slice;
+    peers = endpoints?.flatMap(e => `${e.addresses}/32`) || [];
+  }
+
+  // Add the clusterIP from the service
   if (k8sApiIP) {
-    peers?.push(k8sApiIP);
+    peers.push(`${k8sApiIP}/32`);
   }
 
-  // If the peers are found, cache and process them
-  if (peers?.length) {
-    apiServerPeers = peers.flatMap(ip => ({
+  // Convert peers into NetworkPolicyPeer objects
+  if (peers.length) {
+    apiServerPeers = peers.flatMap(cidr => ({
       ipBlock: {
-        cidr: `${ip}/32`,
+        cidr,
```

Comment:
do we still need the `cidr: ` prefix here?

Reply:
Since the var is already named `cidr` it "just works", but I can re-add to be explicit.

Reply:
Understood, no need to add it back if this works!